src/components/typography/Heading.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { PropsWithChildren } from 'react';
+
+export enum HeadingSize {
+  ExtraSmall = 'xsmall',
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+  ExtraLarge = 'xlarge',
+}
+
+const bottomMarginByHeadingSize = (headingSize: HeadingSize) => {
+  switch (headingSize) {
+    case HeadingSize.Large:
+      return 'var(--echoes-dimension-space-200)';
+    case HeadingSize.ExtraLarge:
+      return 'var(--echoes-dimension-space-250)';
+    default:
+      return 'var(--echoes-dimension-space-100)';
+  }
+};
+
+type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5';
+
+const defaultSizeByTag: Record<HeadingTag, HeadingSize> = {
+  h1: HeadingSize.ExtraLarge,
+  h2: HeadingSize.Large,
+  h3: HeadingSize.Medium,
+  h4: HeadingSize.Small,
+  h5: HeadingSize.ExtraSmall,
```

Comment:
let's add h6 with ExtraSmall too? To follow the documentation in figma...

Reply:
I only see five sizes on Figma?

Reply:
Yes only 5 sizes but seems like it's supposed to support the h6: 
![image](https://github.com/user-attachments/assets/20463d57-3649-4f5d-8f03-ad43afedc0b3)

Reply:
I'm not sure how we should interpret this description: it could mean that, where we used either `h5` or `h6` before, we should now use `xsmall` consistently, whatever actual tag ends up being used? Let's ask @marciopmoreira6

Reply:
but if they want to specifically use `h6` to mean "not `h5`, something smaller," hard-coding a behavior for `h6` that is the same as `h5` won't allow them to do what they want, either. What we really want to know is if, from a design point of view, we *should* have both `h5` and `h6`, as two different things

Reply:
I think he would have added an xxs heading specifically for h6 if we wanted a different style for h6.

Reply:
as per our conversation with @marciopmoreira6, we only allow h1 through h5, and if we ever see h6 in the products we replace it with h5.